EDF.py

```diff
@@ -44,105 +89,155 @@ def close(self):
         # this requires copying the whole file content
         meas_info = self.meas_info
         chan_info = self.chan_info
+
         # update the n_records value in the file
         tempname = self.fname + '.bak'
         os.rename(self.fname, tempname)
+
         with open(tempname, 'rb') as fid1:
             assert(fid1.tell() == 0)
             with open(self.fname, 'wb') as fid2:
                 assert(fid2.tell() == 0)
                 fid2.write(fid1.read(236))
-                fid1.read(8)                                    # skip this part
-                fid2.write(padtrim(str(self.n_records), 8))     # but write this instead
-                fid2.write(fid1.read(meas_info['data_offset'] - 236 - 8))
-                blocksize = np.sum(chan_info['n_samps']) * meas_info['data_size']
+
+                # skip this part
+                fid1.read(8)
+                # but write this instead
+                writebyte(fid2, padtrim(str(self.n_records), 8))
+
+                writebyte(fid2, fid1.read(
+                    meas_info['data_offset'] - 236 - 8))
+                
+                blocksize = np.sum(
+                    chan_info['n_samps']) * meas_info['data_size']
+                
                 for block in range(self.n_records):
-                    fid2.write(fid1.read(blocksize))
+                    writebyte(fid2, fid1.read(blocksize))
+        
         os.remove(tempname)
-        self.fname = None
-        self.meas_info = None
-        self.chan_info = None
-        self.calibrate = None
-        self.offset    = None
-        self.n_records = 0
-        return
+        self.init_properties()
```

Comment:
The original lines did not handle fname in close(). Hence I wrapped the repetitive lines like this. I wanted to rename the function properly but failed to find a better name. Afterwards, I got busy in office works. Do you have any suggestions?

Reply:
@palday @robertoostenveld I just thought about renaming the function as initial_prop() or initial_state_props (influenced from ReactJS). This way, we can address that the function is supposed to set a few object properties to their initial state, whenever invoked. This, i hope, will end any probable confusion of the function purpose for construction and reset of the objects.

Also, the function is created as a method for each object, causing repetition. What if we create it as a global function, or as under an object that only hosts this property which can be inherited by both edfreader and edfwriter? This may seem unnecessary to you and please let me know if so. I am just having ideas.